src/node_quic_crypto.cc

```diff
@@ -1804,5 +805,121 @@ Local<Value> GetCipherVersion(Environment* env, SSL* ssl) {
   return version;
 }
 
+namespace {
+int SetEncryptionSecrets(
+    SSL* ssl,
+    OSSL_ENCRYPTION_LEVEL ossl_level,
+    const uint8_t* read_secret,
+    const uint8_t* write_secret,
+    size_t secret_len) {
+  QuicSession* session = static_cast<QuicSession*>(SSL_get_app_data(ssl));
+  return session->OnSecrets(
+      from_ossl_level(ossl_level),
+      read_secret,
+      write_secret,
+      secret_len) ? 1 : 0;
+}
+
+int AddHandshakeData(
+    SSL* ssl,
+    OSSL_ENCRYPTION_LEVEL ossl_level,
+    const uint8_t* data,
+    size_t len) {
+  QuicSession* session = static_cast<QuicSession*>(SSL_get_app_data(ssl));
+  session->WriteHandshake(from_ossl_level(ossl_level), data, len);
+  return 1;
+}
+
+int FlushFlight(SSL *ssl) { return 1; }
+
+int SendAlert(
+    SSL* ssl,
+    enum ssl_encryption_level_t level,
+    uint8_t alert) {
+  QuicSession* session = static_cast<QuicSession*>(SSL_get_app_data(ssl));
+  session->SetTLSAlert(alert);
+  return 1;
+}
+
+SSL_QUIC_METHOD quic_method = SSL_QUIC_METHOD{
+  SetEncryptionSecrets,
+  AddHandshakeData,
+  FlushFlight,
+  SendAlert
+};
+}  // namespace
+
+void SetQuicMethod(SSL_CTX* ctx) {
+  SSL_CTX_set_quic_method(ctx, &quic_method);
+}
+
+namespace {
+void SetALPN(SSL* ssl, const std::string& alpn) {
+  SSL_set_alpn_protos(
+      ssl,
+      reinterpret_cast<const uint8_t*>(alpn.c_str()),
+      alpn.length());
+}
+
+void SetHostname(SSL* ssl, const std::string& hostname) {
+  if (hostname.length() == 0 ||
+      SocketAddress::numeric_host(hostname.c_str())) {
+    SSL_set_tlsext_host_name(ssl, "localhost");
```

Comment:
Is it intentional that all IP addresses are implicitly set to `localhost`?

Reply:
Ah, that's temporary, just forgot to add a `TODO` for that.